component-form/component-server-proxy-parent/component-server-proxy/src/main/java/org/talend/sdk/component/proxy/service/ActionService.java

```diff
@@ -157,14 +157,22 @@
                 params.entrySet().stream().filter(it -> it.getKey().endsWith(".$selfReference")).map(it -> {
                     final String configType =
                             ofNullable(params.get(it.getKey() + "Type")).map(String::valueOf).orElse(null);
-                    return referenceService.findPropertiesById(configType, it.getKey(), context).thenApply(form -> {
-                        final String prefix = it.getKey().replace(".$selfReference", ".");
-                        synchronized (newProps) {
-                            ofNullable(form.getProperties())
-                                    .ifPresent(props -> props.forEach((k, v) -> newProps.put(prefix + k, v)));
-                        }
-                        return form;
-                    });
+                    return referenceService
+                            .findPropertiesById(configType, String.valueOf(it.getValue()), context)
+                            .thenApply(form -> {
+                                final String prefix = it.getKey().replace(".$selfReference", "");
+                                synchronized (newProps) {
+                                    ofNullable(form.getProperties())
+                                            .ifPresent(props -> props
+                                                    .entrySet()
+                                                    .stream()
+                                                    .filter(k -> k.getKey().indexOf(".") > 0)
```

Comment:
is there any other way to filter $formId and find the root path properly?

Reply:
yep, idea is to find the form spec (simple property definition list) from the $formId of the nested config (form) and find the property where name==path

@akhabali is looking that since he is blocked by that as well